cpp/src/arrow/util/value_parsing.h

```diff
@@ -561,6 +562,38 @@ static inline bool ParseTimestampStrptime(const char* buf, size_t length,
   return true;
 }
 
+/// \brief Returns time since the UNIX epoch in the requested unit. Takes null terminated
+/// buffer as argument
+static inline bool ParseTimestampStrptime(const char* buf, const char* format,
+                                          bool ignore_time_in_day,
+                                          bool allow_trailing_chars, TimeUnit::type unit,
+                                          int64_t* out) {
+  // NOTE: strptime() is more than 10x faster than arrow_vendored::date::parse().
+  struct tm result;
+  memset(&result, 0, sizeof(struct tm));
+#ifdef _WIN32
+  char* ret = arrow_strptime(buf, format, &result);
+#else
+  char* ret = strptime(buf, format, &result);
+#endif
+  if (ret == NULLPTR) {
+    return false;
+  }
+  if (!allow_trailing_chars && static_cast<size_t>(ret - buf) != strlen(buf)) {
```

Comment:
we can save the strlen if you pass length as argument.

Reply:
done